glommio/src/channels/shared_channel.rs

```diff
@@ -762,4 +762,179 @@ mod test {
         ex1.join().unwrap();
         ex2.join().unwrap();
     }
+
+    #[derive(Debug)]
+    struct WithDrop(Arc<AtomicUsize>, usize);
+
+    impl Drop for WithDrop {
+        fn drop(&mut self) {
+            self.0.fetch_sub(1, Ordering::Relaxed);
+        }
+    }
+
+    #[test]
+    fn shared_drop_gets_called() {
+        let (sender, receiver) = new_bounded(1000);
+
+        let original = Arc::new(AtomicUsize::new(0));
+        let send_count = original.clone();
+        let drop_count = original.clone();
+
+        let ex1 = LocalExecutorBuilder::new()
+            .spawn(move || async move {
+                let sender = sender.connect().await;
+                for x in 0..1000 {
+                    let val = WithDrop(send_count.clone(), x);
+                    drop_count.fetch_add(1, Ordering::Relaxed);
+                    match sender.send(val).await {
+                        Ok(_) => {}
+                        Err(_) => {
+                            // dbg!(err);
```

Comment:
if this is an unexpected condition we should panic. Tests should be extremely reliable and as few surprises as possible

Reply:
Oh the errors are actually OK in these tests because we aren't interested in if it's closed or not yet, we just wanna make sure everything drops, no matter where the items are at the time of the drops. So I'll clean up those match statements cause they take up too much space.